File: backend/check_bucket.py
import os
from supabase import create_client, Client
from dotenv import load_dotenv

load_dotenv()
supabase = create_client(os.environ.get("SUPABASE_URL"), os.environ.get("SUPABASE_KEY"))

try:
    buckets = supabase.storage.list_buckets()
    bucket_names = [b.name for b in buckets]
    if "property-images" not in bucket_names:
        print("Creating property-images bucket...")
        supabase.storage.create_bucket("property-images", {"public": True})
        print("Bucket created!")
    else:
        print("Bucket already exists.")
except Exception as e:
    print("Error:", e)

File: frontend/check_bucket.js
import { createClient } from '@supabase/supabase-js';
import dotenv from 'dotenv';
dotenv.config();

const supabase = createClient(process.env.VITE_SUPABASE_URL, process.env.VITE_SUPABASE_KEY);

async function check() {
    const { data: buckets } = await supabase.storage.listBuckets();
    if (!buckets.find(b => b.name === 'property-images')) {
        console.log("Creating bucket...");
        const { data, error } = await supabase.storage.createBucket('property-images', { public: true });
        if (error) console.log(error);
        else console.log("Created successfully");
    } else {
        console.log("Bucket already exists");
    }
}
check();
